client/packages/invoices/src/Prescriptions/LineEditView/PrescriptionLineEdit.tsx

```diff
@@ -1,139 +1,127 @@
-    if (
-      // Don't allow save if both prescribed quantity and allocated packs are zero for all lines
-      newAllocateQuantities?.every(
-        line =>
-          line.numberOfPacks === 0 &&
-          (!line.prescribedQuantity || line.prescribedQuantity === 0)
-      )
-    ) {
-      setIsDirty(false);
+  useEffect(() => {
+    if (!currentItemId) {
+      clear(); // Clear context if switched to new item
+      return;
     }
 
-    return newAllocateQuantities;
-  };
-
-  const canAutoAllocate = !!(currentItem && draftPrescriptionLines.length);
+    // Manual query for item + prescription line data,
+    // only initialise for allocation when data is available
+    queryData().then(({ data }) => {
+      if (!data) return;
 
-  const hasOnHold = draftPrescriptionLines.some(
-    ({ stockLine }) =>
-      (stockLine?.availableNumberOfPacks ?? 0) > 0 && !!stockLine?.onHold
-  );
-  const hasExpired = draftPrescriptionLines.some(
-    ({ stockLine }) =>
-      (stockLine?.availableNumberOfPacks ?? 0) > 0 &&
-      !!stockLine?.expiryDate &&
-      DateUtils.isExpired(new Date(stockLine?.expiryDate))
-  );
+      initialise({
+        itemData: data,
+        strategy: sortByVvmStatus
+          ? AllocationStrategy.VVMStatus
+          : AllocationStrategy.FEFO,
+        allowPlaceholder: false,
+        allowPrescribedQuantity: true,
+        ignoreNonAllocatableLines: true,
+        // In prescriptions, default to allocate in doses for vaccines
+        // if pref is on
+        allocateIn:
+          allocateVaccineItemsInDoses && data.item.isVaccine
+            ? { type: AllocateInType.Doses }
+            : undefined,
+      });
+    });
+    // Expect dependencies to be stable
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [currentItemId]);
 
   return (
-    <PrescriptionLineEditForm
-      disabled={isDisabled}
-      isNew={isNew}
-      packSizeController={packSizeController}
-      onChangeItem={(item: ItemRowWithDirectionsFragment | null) => {
-        setIsAutoAllocated(false);
-        setCurrentItem(item);
-      }}
-      item={currentItem}
-      allocatedUnits={getAllocatedQuantity(draftPrescriptionLines)}
-      availableUnits={sumAvailableQuantity(draftPrescriptionLines)}
-      onChangeQuantity={onAllocate}
-      canAutoAllocate={canAutoAllocate}
-      isAutoAllocated={isAutoAllocated}
-      updateNotes={onUpdateNotes}
-      draftPrescriptionLines={draftPrescriptionLines}
-      showZeroQuantityConfirmation={showZeroQuantityConfirmation}
-      hasOnHold={hasOnHold}
-      hasExpired={hasExpired}
-      isLoading={isLoading}
-      updateQuantity={onUpdateQuantity}
-      programId={programId}
-    />
+    <Grid
+      container
+      gap="4px"
+      sx={{ minHeight: 200, display: 'flex', flexDirection: 'column' }}
+    >
+      <AccordionPanelSection
+        // Key ensures component will reload when switching item, but not when
```

Comment:
Is there meant be a key here @lache-melvin ?

Reply:
whoops, moved a layer out, have removed stale comments [6f93704](https://github.com/msupply-foundation/open-msupply/pull/7888/commits/6f93704b342cc694b18706ee57ea070a39c244ef)